http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -950,10 +950,19 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
                                             gkeInterp.getGalaxyPostgresDiskName(dbApp.app.appResources.namespace.name))
         diskDetachResult <- streamUntilDoneOrTimeout(
           getDiskDetachStatus(originalDetachTimestampOpt, getDisk),
-          24,
+          30,
           5 seconds,
           "The disk failed to detach within the time limit, cannot proceed with delete disk"
-        )
+        ).adaptError {
+          case e =>
```

Comment:
If this failed, it threw an uncaught exception that caused the db state for the app not being properly updated to error status.